armadillo/src/main/java/org/molgenis/armadillo/ArmadilloSession.java

```diff
@@ -61,20 +66,31 @@ void tryDetachRConnection(RConnection connection) {
   RConnection getRConnection() {
     try {
       if (rSession == null) {
-        return connectionFactory.createConnection();
+        var connection = connectionFactory.createConnection();
+        pid = processService.getPid(connection);
+        return connection;
       }
-      return rSession.attach();
+      var connection = rSession.attach();
+      rSession = null;
+      return connection;
     } catch (RserveException err) {
       throw new ArmadilloSessionException("Could not attach connection to RSession", err);
     }
   }
 
-  public synchronized void sessionCleanup() {
```

Comment:
No longer needs to wait for evaluation to be finished, it kills the process instead!